### ISSUE TITLE
Add lineages column in `prefetch` csv output

### ISSUE THREAD

**Issue by mr-eyes** [sourmash-bio/sourmash#2472]
Maybe this could be an added option to `tax annotate` or added by default to the `prefetch` output.

**Comment by ctb** [comment]
> Maybe this could be an added option to `tax annotate` or added by default to the `prefetch` output.

note that prefetch does not have access to lineages! So it would have to be done with `tax annotate`.

**Comment by bluegenes** [comment]
**tl;dr** 
To fix, I think we should just make a separate file-reading option for `tax annotate` that only requires the `name` / `match_name` column and does not check for anything else. This would enable `tax annotate` to be extremely flexible, even annotating a file with just the `ident` column and no other info. We would continue to simply add an additional column to each row containing the taxonomic annotation. Thoughts @ctb? Would this fit your usage @mr-eyes?

**details:**

Since prefetch columns are slightly different than gather columns, this does not currently work. Here, I tested using a prefetch file in place of a gather file. 
`sourmash tax annotate -g ONT_R10_Zymo_D6300.dna-k31-sc1000.prefetch.csv -t /group/ctbrowngrp/sourmash-db/genbank-2022.03/genbank-2022.03-lineages.csv`


Error from v4.6.1:
```
== This is sourmash version 4.6.1. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

Exiting.
Traceback (most recent call last):
  File "/home/ntpierce/miniconda3/envs/sourmash/bin/sourmash", line 11, in <module>
    sys.exit(main())
  File "/home/ntpierce/miniconda3/envs/sourmash/lib/python3.10/site-packages/sourmash/__main__.py", line 13, in main
    return mainmethod(args)
  File "/home/ntpierce/miniconda3/envs/sourmash/lib/python3.10/site-packages/sourmash/cli/tax/annotate.py", line 67, in main
    return sourmash.tax.__main__.annotate(args)
  File "/home/ntpierce/miniconda3/envs/sourmash/lib/python3.10/site-packages/sourmash/tax/__main__.py", line 338, in annotate
    gather_results, idents_missed, total_missed, header = tax_utils.check_and_load_gather_csvs(g_csv, tax_assign, force=args.force,
  File "/home/ntpierce/miniconda3/envs/sourmash/lib/python3.10/site-packages/sourmash/tax/tax_utils.py", line 140, in check_and_load_gather_csvs
    these_results, header, seen_queries = load_gather_results(gather_csv, seen_queries=seen_queries, force=force)
  File "/home/ntpierce/miniconda3/envs/sourmash/lib/python3.10/site-packages/sourmash/tax/tax_utils.py", line 104, in load_gather_results
    raise ValueError(f"Not all required gather columns are present in '{gather_csv}'.")
ValueError: Not all required gather columns are present in 'ONT_R10_Zymo_D6300.dna-k31-sc1000.prefetch.csv'.
```
This code has changed a bit from the release to `latest`.

with `latest`, that error will come from here: https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/tax/tax_utils.py#L425 and is slightly more complicated because it finds an additional column, `jaccard`, which we're not expecting.

```
== This is sourmash version 4.6.1. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

Exiting.
Traceback (most recent call last):
  File "/Users/ntward/dib-lab/sourmash/src/sourmash/tax/tax_utils.py", line 425, in load_gather_results
    gatherRow = GatherRow(**row)
TypeError: __init__() got an unexpected keyword argument 'jaccard'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ntward/mambaforge/envs/sourmash_dev/bin/sourmash", line 8, in <module>
    sys.exit(main())
  File "/Users/ntward/dib-lab/sourmash/src/sourmash/__main__.py", line 13, in main
    return mainmethod(args)
  File "/Users/ntward/dib-lab/sourmash/src/sourmash/cli/tax/annotate.py", line 67, in main
    return sourmash.tax.__main__.annotate(args)
  File "/Users/ntward/dib-lab/sourmash/src/sourmash/tax/__main__.py", line 288, in annotate
    query_gather_results = tax_utils.check_and_load_gather_csvs(gather_csvs, tax_assign, force=args.force,
  File "/Users/ntward/dib-lab/sourmash/src/sourmash/tax/tax_utils.py", line 465, in check_and_load_gather_csvs
    these_results, header = load_gather_results(gather_csv, tax_assign,
  File "/Users/ntward/dib-lab/sourmash/src/sourmash/tax/tax_utils.py", line 427, in load_gather_results
    raise ValueError(f"'{gather_csv}' is missing columns needed for taxonomic summarization. Please run gather with sourmash >= 4.4.") from exc
ValueError: 'ONT_R10_Zymo_D6300.dna-k31-sc1000.prefetch.csv' is missing columns needed for taxonomic summarization. Please run gather with sourmash >= 4.4.
```

For taxonomic summarization, we require the following essential columns:

```
    # essential columns
    query_name: str
    name: str # match_name
    f_unique_weighted: float
    f_unique_to_query: float
    unique_intersect_bp: int
    remaining_bp: int
    query_md5: str
    query_filename: str
    # new essential cols: requires 4.4x
    query_bp: int
    ksize: int
    scaled: int
```
In `prefetch` results, `name` is replaced with `match_name` and we don't have  `f_unique_weighted`,`f_unique_to_query`, `unique_intersect_bp`, or `remaining_bp` because there was no gather subtraction done. We also have some additional columns which are not in `gather` results.

**Comment by ctb** [comment]
> **tl;dr**
> To fix, I think we should just make a separate file-reading option for `tax annotate` that only requires the `name` / `match_name` column and does not check for anything else. This would enable `tax annotate` to be extremely flexible, even annotating a file with just the `ident` column and no other info. We would continue to simply add an additional column to each row containing the taxonomic annotation. Thoughts @ctb? Would this fit your usage @mr-eyes?

yes I think that works!

**Comment by mr-eyes** [comment]
> Would this fit your usage @mr-eyes?

Thanks, Tessa! I am gonna try this and update the issue. I don't know if opening a PR with that would be a good idea.
